src/launcher/launcher.sh

```diff
@@ -0,0 +1,78 @@
+# Copyright The async-profiler authors
+# SPDX-License-Identifier: Apache-2.0
+
+#!/bin/sh
+
+set -euo pipefail
+
+VERSION_STRING="JFR converter PROFILER_VERSION built on BUILD_DATE"
+
+if [[ "$1" == "-v" || "$1" == "--version" ]]; then
+    echo "$VERSION_STRING"
+    exit 0
+fi
+
+SCRIPT_PATH="$0"
+
+JAVA_CMD=("-Xss2M" "-Dsun.misc.URLClassPath.disableJarChecking")
+
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -D*|-X*|-agent*)
+            if [[ ${#1} -gt 2 ]]; then
+                JAVA_CMD+=("$1")
+            fi
+            ;;
+        -J*)
+            JAVA_CMD+=("${1:2}")
+            ;;
+        *)
+            break
+            ;;
+    esac
+    shift
+done
+
+JAVA_CMD+=("-jar" "$SCRIPT_PATH" "$@")
+
+run_java() {
```

Comment:
Is there a reason to put it in a function, if the following code is executed as is anyway?
E.g., the above argument parsing logic is not in a function, and that's fine.